Added variable lint.

richardgaunt opened the pull request:
## Checklist before requesting a review

- [x] This PR does not have an associated ticket
- [x] I have provided information in  section about WHY something was done if this was not a normal implementation
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run new and existing relevant tests locally with my changes, and they passed
- [ ] I have provided screenshots, where applicable

## Changed
1. Added lint to check variables completeness.

## Screenshots


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Chores**
  * Added automated validators that detect unpaired light/dark theme variables and unused CSS custom properties, with readable reports and non‑zero exit on failures.

* **Style**
  * Simplified theme defaults: removed the base animation duration and consolidated many stripe tokens to use CSS custom properties.
  * Removed generation of "visited" variant styles across components.
  * Updated CSS variable emission to omit a navigation orientation threshold variable.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

alan-cole commented:
@richardgaunt - Great idea! I've gone and fixed the duplicates found, and then realized the missing variables were because those variables were no longer used and hadn't been fully removed for both light and dark.

Which made me think we need a script to also pick up on any unused variables, so I've gone and added one to this PR, as well as remove all the unused variables too.

richardgaunt commented:
> @richardgaunt - Great idea! I've gone and fixed the duplicates found, and then realized the missing variables were because those variables were no longer used and hadn't been fully removed for both light and dark.
> 
> Which made me think we need a script to also pick up on any unused variables, so I've gone and added one to this PR, as well as remove all the unused variables too.

Fantastic additions makes this 100% better
